tests/integration/testlib/statelib.py

```diff
@@ -77,7 +78,15 @@ def filter(self, based_on_state):
             for ifstate in self._state[INTERFACES]
             if ifstate['name'] in base_iface_names
         ]
-        self._state = {INTERFACES: filtered_iface_state}
+        for rt_type in (Route.RUNNING, Route.CONFIG):
+            filtered_routes = [
+                route
+                for route in self._state.get(Route.KEY, {}).get(rt_type, [])
+                if route.get(Route.NEXT_HOP_INTERFACE) in base_iface_names
+            ]
+            if filtered_routes:
+                self._state[Route.KEY][rt_type] = filtered_routes
+        self._state[INTERFACES] = filtered_iface_state
```

Comment:
It is confusing that this line comes after the route filtering. IMHO it should stay before the route filtering. Also the change from not using `INTERFACES: ...` to the new form would have been a perfect change for a commit with the related changes to use `netinfo.show()`.